membership-attribute-service/app/models/Attributes.scala

```diff
@@ -19,14 +17,36 @@ object ContentAccess {
   implicit val jsWrite = Json.writes[ContentAccess]
 }
 
+case class CardDetails(last4: String, expirationMonth: Int, expirationYear: Int, forProduct: String) {
+  val asLocalDate = new LocalDate(expirationYear, expirationMonth, 1).plusMonths(1).minusDays(1)
```

Comment:
My main reason for preferring a def here is because this `case class` is used for json serialisation.  If the month/year do not represent a valid date then json deserialisation will fail with an exception and I'm not sure this will be surfaced in an informative way.